File: sdk/java/src/test/java/com/flagent/client/model/EvalDebugLogTest.java
package com.flagent.client.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for EvalDebugLog
 */
class EvalDebugLogTest {

    @Test
    void testEvalDebugLog() {
        SegmentDebugLog segLog = new SegmentDebugLog().segmentID(1L).msg("segment matched");
        List<SegmentDebugLog> segmentLogs = Collections.singletonList(segLog);
        EvalDebugLog model = new EvalDebugLog()
            .msg("Evaluation completed")
            .segmentDebugLogs(segmentLogs);
        Assertions.assertNotNull(model);
        Assertions.assertEquals("Evaluation completed", model.getMsg());
        Assertions.assertEquals(1, model.getSegmentDebugLogs().size());
        Assertions.assertEquals("segment matched", model.getSegmentDebugLogs().get(0).getMsg());
    }

    @Test
    void msgTest() {
        EvalDebugLog model = new EvalDebugLog().msg("message");
        Assertions.assertEquals("message", model.getMsg());
    }

    @Test
    void segmentDebugLogsTest() {
        List<SegmentDebugLog> logs = new ArrayList<>();
        logs.add(new SegmentDebugLog().segmentID(1L).msg("m1"));
        EvalDebugLog model = new EvalDebugLog().segmentDebugLogs(logs);
        Assertions.assertEquals(1, model.getSegmentDebugLogs().size());
        Assertions.assertEquals(1L, model.getSegmentDebugLogs().get(0).getSegmentID());
    }

}

File: backend/src/test/kotlin/flagent/service/EvalDebugLogTest.kt
package flagent.service

import kotlin.test.Test
import kotlin.test.assertEquals

class EvalDebugLogTest {
    @Test
    fun `create EvalDebugLog with all fields`() {
        val segmentLogs = listOf(
            SegmentDebugLog(segmentID = 1, message = "Segment matched"),
            SegmentDebugLog(segmentID = 2, message = "Segment not matched")
        )
        
        val debugLog = EvalDebugLog(
            message = "Evaluation completed",
            segmentDebugLogs = segmentLogs
        )
        
        assertEquals("Evaluation completed", debugLog.message)
        assertEquals(2, debugLog.segmentDebugLogs.size)
        assertEquals(1, debugLog.segmentDebugLogs[0].segmentID)
        assertEquals("Segment matched", debugLog.segmentDebugLogs[0].message)
    }
    
    @Test
    fun `create EvalDebugLog with default values`() {
        val debugLog = EvalDebugLog()
        
        assertEquals("", debugLog.message)
        assertEquals(0, debugLog.segmentDebugLogs.size)
    }
    
    @Test
    fun `EvalDebugLog data class equality`() {
        val log1 = EvalDebugLog(
            message = "Test",
            segmentDebugLogs = listOf(SegmentDebugLog(1, "Message"))
        )
        val log2 = EvalDebugLog(
            message = "Test",
            segmentDebugLogs = listOf(SegmentDebugLog(1, "Message"))
        )
        
        assertEquals(log1, log2)
        assertEquals(log1.hashCode(), log2.hashCode())
    }
}
